src/handlers/gestures/simpleGestures.ts

```diff
@@ -0,0 +1,381 @@
+  };
+
+  //TODO fix type
+  static allowedProps: any = baseGestureHandlerWithMonitorProps;
+
+  constructor() {
+    super();
+  }
+
+  protected setConfig(key: string, value: any) {
+    this.config[key] = value;
+  }
+
+  protected setHandler(key: string, value: any) {
+    this.handlers[key] = value;
+  }
+
+  private addDependency(
+    key: string,
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    if (this.config[key]) {
+      if (Array.isArray(this.config[key])) {
+        this.config[key].push(gesture);
+      } else {
+        this.config[key] = [this.config[key], gesture];
+      }
+    } else {
+      this.config[key] = [gesture];
+    }
+  }
+
+  setRef(ref: React.RefObject<any>) {
+    this.setConfig('ref', ref);
+    return this;
+  }
+
+  setOnBegan(callback: (event: any) => void) {
+    this.setHandler('onBegan', callback);
+    return this;
+  }
+
+  setOnStart(callback: (event: any) => void) {
+    this.setHandler('onStart', callback);
+    return this;
+  }
+
+  setOnEnd(callback: (event: any, success: boolean) => void) {
+    this.setHandler('onEnd', callback);
+    return this;
+  }
+
+  setOnUpdate(callback: (event: any) => void) {
+    this.setHandler('onUpdate', callback);
+    return this;
+  }
+
+  setEnabled(enabled: boolean) {
+    this.setConfig('enabled', enabled);
+    return this;
+  }
+
+  setMinPointers(minPointers: number) {
+    this.setConfig('minPointers', minPointers);
+    return this;
+  }
+
+  setShouldCancelWhenOutside(value: boolean) {
+    this.setConfig('shouldCancelWhenOutside', value);
+    return this;
+  }
+
+  setHitSlop(hitSlop: any) {
+    this.setConfig('hitSlop', hitSlop);
+    return this;
+  }
+
+  addSimultaneousGesture(
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    this.addDependency('simultaneousWith', gesture);
+
+    return this;
+  }
+
+  addRequiredToFailGesture(
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    this.addDependency('requireToFail', gesture);
+
+    return this;
+  }
+
+  simultaneousWith(other: SimpleGesture): GestureBuilder {
+    return new GestureBuilder(this).simultaneousWith(other);
+  }
+
+  exclusiveWith(other: SimpleGesture): GestureBuilder {
+    return new GestureBuilder(this).exclusiveWith(other);
+  }
+
+  requireToFail(other: SimpleGesture): GestureBuilder {
+    return new GestureBuilder(this).requireToFail(other);
+  }
+
+  initialize() {
+    this.handlerTag = getNextHandlerTag();
+    this.handlers = { ...this.handlers, handlerTag: this.handlerTag };
+
+    if (this.config.ref) {
+      this.config.ref.current = this;
+    }
+  }
+
+  build(): BuiltGesture {
+    let result = new BuiltGesture(this.prepare);
+
+    result.gestures = [this];
+
+    return result;
+  }
+
+  prepare = () => {
+    if (
+      this.config.requireToFail &&
+      !Array.isArray(this.config.requireToFail)
+    ) {
+      this.config.requireToFail = [this.config.requireToFail];
+    }
```

Comment:
```js
private toArray(x: unknown) {
	return [].concat(x);	
}

const config = this.config;

if (config.requireToFail !== undefined) {
	config.requireToFail = toArray(config.requireToFail);
}

// ...
```

Reply:
Added in c110e1793d3e100c87080f9996d32922da44daef